content/blog/2022/2022-05-22-connect-to-aws-rds.md

```diff
@@ -0,0 +1,120 @@
+url: connect-rds-byjumphost
+---
+
+Back-end server resources like databases often contain data that is critical for an application to function consistently. So these resources are protected from public access over the internet by placing them in a private subnet. This will however make it inaccessible to the database clients and applications running on our local development workstations. 
+
+This problem is addressed by using a server called "Jump host" that can receive requests from external sources over the internet and securely forward or "jump" to the database secured in the private subnet.
+
+In this tutorial, we will use a jump host for accessing an Amazon Relational Database Service(RDS) database residing in a private subnet.
+
+## Creating an RDS Database with Engine Type: MySQL
+Let us first create our RDS database using the AWS Management Console with `MySQL` as the engine type:
+
+{{% image alt="Create RDS Database" src="images/posts/aws-rds-connect/create-db.png" %}}
+
+For creating the RDS database in a private subnet we have used the following configurations:
+{{% image alt="Create RDS Database" src="images/posts/aws-rds-connect/connect.png" %}}
+
+We have used the default virtual private cloud(VPC) available in our AWS account and set the `public access` to `No`. We have also chosen the option to create a new security group named `db-sg` where we will define the inbound rules to allow traffic from selected sources. 
+
+We will also select `Password authentication` as the Database authentication option.
+
+Our RDS database created in a private subnet is ready to use when the status changes to `available`
+{{% image alt="Create RDS Database" src="images/posts/aws-rds-connect/db-created.png" %}}
+
+When the database is ready to be used, we can see the endpoint of the database along with the port which we will use later to connect to the database.
+
+With our database created, we will next set up a jump host and populate inbound rules in the security groups in the following sections.
+
+## Creating an EC2 Instance as the Jump Host 
+A jump host is also called a bastion host/server whose sole purpose is to provide access to resources in a private network from an external network like the internet. A rough representation of this architecture is shown below:
+
+{{% image alt="Jump host" src="images/posts/aws-rds-connect/jump-host.png" %}}
+
+Here we are using an EC2 instance in a public subnet as our jump host for connecting to an RDS database in a private subnet.
+
+Let us create the EC2 instance from the AWS Management Console in a public subnet in the same VPC where we had created our RDS database in the previous section.:
+
+{{% image alt="Create EC2 bastion" src="images/posts/aws-rds-connect/create-ec2.png" %}}
+
+We have created our instance in the free tier with an SSH key pair to access the instance with SSH in the later sections. The private key of the SSH key pair is downloaded and saved to our local workstation. 
```

Comment:
Is the SSH key pair created automatically, or we should do anything actively? Like clicking on a checkbox?